### PR TITLE
Feat/appl records/v28

### PR DESCRIPTION
**Pull request by victorjulien**
Draft branch for app records.

suricata-verify-pr: 592

### CODE REVIEW

**Comment by victorjulien** [comment]
replaces #6387

**Comment by victorjulien** [comment]
Replacing the asserts with BUG_ONs.

Improving keyword parsing and validation.

File: suricata.yaml.in [hunk 1]
```diff
@@ -165,6 +165,10 @@ outputs:
             # Enable the logging of tagged packets for rules using the
             # "tag" keyword.
             tagged-packets: yes
+        # app layer records
+        - record:
+            # TODO disable by default. Too noisy.
+            enabled: yes
```
Comment (jasonish) [inline suricata.yaml.in:171]: Not really specific to this, but I wonder if we should name it "debug:record" (or `debug-record` if the `:` doesn't work), just to make it clear.  We'll come across this case again where an output is useful for debugging/dev, but really shouldn't be turned on by most end-users in production.

Comment (victorjulien) [reply suricata.yaml.in:171]: Perhaps ya, though I'd also see it as useful for rule writers/analysts, so the name shouldn't be too dev-y

**Comment by victorjulien** [comment]
Cleaning up commits. Next PR should be easier to review

**Comment by victorjulien** [comment]
Replaced by #6665

**Pull request closed without merging**